Dockerfile.dapper

```diff
@@ -1,9 +1,9 @@
-FROM golang:1.16-alpine
+FROM registry.suse.com/bci/golang:1.17
```

Comment:
One thing that I want to point out here - this is also upgrading the go version used in build from 1.16 to 1.17.

The build succeeded (and we were probably going to need to do this at some point anyway), but you will probably want to note this so that QA can do a smoke test of this and make sure (at a high level) that nothing broke.

There's also a go version that is specified in the [go.mod](https://github.com/rancher/kube-api-auth/blob/21cedd184a62d77c7820e7bf5443be45348ca90f/go.mod#L3), but keep in mind that according to the [go docs](https://go.dev/doc/modules/gomod-ref) that this is a minimum version, so I don't think that this is necessary to change at this time.